lib/spack/spack/compilers/__init__.py

```diff
@@ -633,7 +633,7 @@ def _default(search_paths):
                         compiler_id = CompilerID(operating_system, compiler_name, None)
                         detect_version_args = DetectVersionArgs(
                             id=compiler_id,
-                            variation=NameVariation(*match.groups()),
+                            variation=NameVariation(*match.groups()[:2]),
```

Comment:
Why limit this?

Reply:
Because before I added the regex pointed out in the comment above this one, the `ext = r"\.(exe|bat)"` there were only two groups being captured, and these were relevant to the construction of that named tuple. This regex adds a new group that we don't care about, so we drop it off. Not doing this causes any code that crosses that line to fail as the group unpacking provides too many positional arguments to the named tuple constructor.

Reply:
I think I get it: I think this should either:

* add a comment here explaining this
* Use the non-capturing group syntax `(?:.....)` described at https://docs.python.org/3/library/re.html when looking for `bat/exe`

(I prefer the latter)

Reply:
I also prefer the latter, forgot it was part of the re syntax. I'll amend.